CovidWatch iOS/Firestore/SignedReportsUploader.swift

```diff
@@ -50,29 +50,51 @@ open class SignedReportsUploader: NSObject, NSFetchedResultsControllerDelegate {
             os_log("Uploading signed report (%@)...", log: .app, signatureBytesBase64EncodedString)
             signedReport.uploadState = UploadState.uploading.rawValue
 
-            self.db.collection(Firestore.Collections.signedReports).addDocument(data: [
-                Firestore.Fields.temporaryContactKeyBytes : signedReport.temporaryContactKeyBytes ?? Data(),
-                Firestore.Fields.endIndex : signedReport.endIndex,
-                Firestore.Fields.memoData : signedReport.memoData ?? Data(),
-                Firestore.Fields.memoType : signedReport.memoType,
-                Firestore.Fields.reportVerificationPublicKeyBytes : signedReport.reportVerificationPublicKeyBytes ?? Data(),
-                Firestore.Fields.signatureBytes : signedReport.signatureBytes ?? Data(),
-                Firestore.Fields.startIndex : signedReport.startIndex,
-                Firestore.Fields.timestamp: FieldValue.serverTimestamp()
-            ]) { [weak self] error in
-                guard let self = self else { return }
-                defer {
-                    try? self.fetchedResultsController.managedObjectContext.save()
-                }
-                
-                if let error = error {
-                    // TODO: Handle retry
-                    os_log("Uploading signed report (%@) failed: %@", log: .app, type: .error, signatureBytesBase64EncodedString, error as CVarArg)
+            // get url to submit to
+            let apiUrlString = getAPIUrl(getAppScheme())
+            if let submitReportUrl = URL(string: "\(apiUrlString)/submitReport") {
+                // build correct payload
+                let reportUpload = ReportUpload(
```

Comment:
There has been lots of discussion in the TCN slack workspace about data formats for uploads. I know initially I thought parsing it out would be nice and intuitive, but now I'm thinking it might make more sense to just simply POST the actual TCN report, especially because the raw report can be fetched by clients and the signature can be verified. Another plus, it matches the TCN protocol description so should be compatible with multiple implementations. What do you think?

Reply:
Yes, I think we should look at changing this when we get time. Currently we are shooting for an MVP this week and the implementations we have already read directly from the firebase. I spoke with Zsombor and in the interest of time were going to just push ahead with this version. Changing to REST lets us disable public writes on the firebase so that people cant just go in and update or delete records. Medium term we will be discussing what the Fairwinds team is doing and possibly just jump straight to an AG (Apple Google) PPCT Protocol compatible API.

Reply:
Okay, we are drafting TCN coalition docs for API and interop here: https://github.com/TCNCoalition/TCN/blob/main/interoperability.md, there is a PR open and more should be coming this week.